src/graphics/sdl/sdl_asset_context.cpp

```
@@ -0,0 +1,35 @@
+#include "canyon/graphics/sdl/sdl_asset_context.h"
+#include "canyon/graphics/sdl/sdl_surface_context.h"
+#include "canyon/graphics/sdl/sdl_font.h"
+#include "canyon/graphics/sdl/sdl_image.h"
+#include "canyon/graphics/sdl/sdl_texture.h"
+
+namespace canyon::graphics::sdl {
+    AssetContext::AssetContext(SurfaceContext& context)
+        : m_context(context) {
+    }
+
+    std::unique_ptr<IImage> AssetContext::NewImage(std::shared_ptr<ITexture> texture) {
```

Clang (14.0.6)

[error] variable 'NewImage' is non-const and globally accessible, consider making it const

(cppcoreguidelines-avoid-non-const-global-variables,-warnings-as-errors)